dev/archery/archery/integration/runner.py

```diff
@@ -0,0 +1,303 @@
+        consumer.stream_to_file(producer_stream_path, consumer_file_path)
+        consumer.validate(json_path, consumer_file_path)
+
+    def _run_gold(self, gold_dir, producer, consumer, test_case):
+        json_path = test_case.path
+
+        # Validate the file
+        print('-- Validating file')
+        producer_file_path = os.path.join(
+            gold_dir, "generated_" + test_case.name + ".arrow_file")
+        consumer.validate(json_path, producer_file_path)
+
+        print('-- Validating stream')
+        consumer_stream_path = os.path.join(
+            gold_dir, "generated_" + test_case.name + ".stream")
+        file_id = guid()[:8]
+        name = os.path.splitext(os.path.basename(json_path))[0]
+
+        consumer_file_path = os.path.join(self.temp_dir, file_id + '_' +
+                                          name + '.consumer_stream_as_file')
+
+        consumer.stream_to_file(consumer_stream_path, consumer_file_path)
+        consumer.validate(json_path, consumer_file_path)
+
+    def _compare_flight_implementations(self, producer, consumer):
+        print('##########################################################')
+        print(
+            '{0} serving, {1} requesting'.format(producer.name, consumer.name)
+        )
+        print('##########################################################')
+
+        for test_case in self.json_files:
+            json_path = test_case.path
+            print('=' * 58)
+            print('Testing file {0}'.format(json_path))
+            print('=' * 58)
+
+            if ('Java' in (producer.name, consumer.name) and
+               "map" in test_case.name):
+                print('TODO(ARROW-1279): Enable map tests ' +
+                      ' for Java and JS once Java supports them and JS\'' +
+                      ' are unbroken')
+                continue
+
+            if SKIP_FLIGHT in test_case.skip:
+                print('-- Skipping test')
+                continue
+
+            try:
+                with producer.flight_server():
+                    # Have the client upload the file, then download and
+                    # compare
+                    consumer.flight_request(producer.FLIGHT_PORT, json_path)
+            except Exception:
+                traceback.print_exc()
+                yield (test_case, producer, consumer, sys.exc_info())
+                continue
+
+
+def get_static_json_files():
+    glob_pattern = os.path.join(ARROW_ROOT_DEFAULT,
+                                'integration', 'data', '*.json')
+    return [
+        datagen.JsonFile(name=os.path.basename(p), path=p, skip=set(),
+                         schema=None, batches=None)
+        for p in glob.glob(glob_pattern)
+    ]
+
+
+def run_all_tests(enable_cpp=True, enable_java=True, enable_js=True,
+                  enable_go=True, run_flight=False,
+                  tempdir=None, **kwargs):
+    tempdir = tempdir or tempfile.mkdtemp()
```

Comment:
mkdtemp should be cleaned up.

Reply:
you mean deleted? As long as there is an option to _not_ delete it (otherwise debugging failures is a lot harder)